backend/lib/endpoints/posts.js

```diff
@@ -374,42 +374,128 @@ async function routes(app) {
+        Comment.aggregate([
+          {
+            $match: {
+              parentId: null,
+              postId: mongoose.Types.ObjectId(postId),
+            },
+          },
+          {
+            $lookup: {
+              as: "children",
+              foreignField: "parentId",
+              from: "comments",
+              localField: "_id",
+            },
+          },
+          {
+            $addFields: {
+              childCount: {
+                $size: { $ifNull: ["$children", []] },
+              },
+            },
+          },
+          {
+            $skip: skip || 0,
+          },
+          {
+            $limit: limit || COMMENT_PAGE_SIZE,
+          },
+        ]),
+      );
+      if (commentErr) {
+        req.log.error("Failed retrieving comments", { commentErr });
+        throw app.httpErrors.internalServerError();
+      }
+
+      return comments;
+    },
+  );
+
+  app.post(
+    "/:postId/comments",
+    { preValidation: [app.authenticate], schema: createCommentSchema },
+    async (req, reply) => {
+      const { parentId, userId } = req.body;
+      const [userErr, user] = await app.to(User.findById(userId));
+      if (userErr) {
+        throw app.httpErrors.notFound();
+      }
+
+      const { postId } = req.params;
+      const { body: commentProps } = req;
+      if (!postId) {
+        throw app.httpErrors.badRequest();
+      }
+
+      // Assign postId and parent comment id (if present)
+      commentProps.postId = postId;
       if (parentId) {
-        const parentPost = await Post.findById(parentId);
-        if (!parentPost || parentPost.postId !== postId) {
-          return new httpErrors.BadRequest();
-        }
+        commentProps.parentId = parentId;
       }
-      return new Comment({
-        ...body,
-        authorId: userId,
-        postId,
-      }).save();
+
+      // Creates embedded author document
+      commentProps.author = {
+        id: user.id,
+        location: user.location,
+        name: user.name,
+        type: user.type,
+      };
+
+      // Initial empty likes array
+      commentProps.likes = [];
+
+      const [err, comment] = await app.to(new Comment(commentProps).save());
+
+      if (err) {
+        req.log.error("Failed creating comment", { err });
+        throw app.httpErrors.internalServerError();
+      }
+
+      reply.code(201);
+      return comment;
     },
   );
 
-  app.put(
+  app.patch(
     "/:postId/comments/:commentId",
     { preValidation: [app.authenticate], schema: updateCommentSchema },
     async (req) => {
-      const { body, params, userId } = req;
-      const { comment } = body;
-      const { commentId, postId } = params;
-      const updatedComment = await Comment.findOneAndUpdate(
-        { _id: commentId, authorId: userId, postId },
-        { comment },
-        { new: true },
-      );
-      if (!updatedComment) {
-        return new httpErrors.BadRequest();
+      const { content, userId } = req.body;
```

Comment:
`userId` comes from JWT here and not the request body (also needs to update the schema)
this is handled by the `authenticate` hook which adds this on `req.userId`

Reply:
Okay, thanks on that. 
I've updated here and it probably needs to be changed on `/posts` endpoint too